stitchee/concatenate.py

```diff
@@ -139,6 +140,104 @@ def validate_concat_method_and_dim(concat_method: str, concat_dim: str | None =
         )
 
 
+def find_variables_with_duplicate_dimensions(dataset: xr.DataSet):
+    # find variables with duplicate dimensions
+    # by comparing dimension's index with its first occurrence
+    duplicate_variables = [
+        varname
+        for varname in dataset.variables
+        if (
+            (dims := dataset[varname].dims)
+            and (any(idim != dims.index(dim) for idim, dim in enumerate(dims)))
+        )
```

Comment:
I think this can be simplified using a `set`:

```suggestion
        if (dims := dataset[varname].dims) and len(dims) != len(set(dims))
```